### PR TITLE
Instrument service members handlers

### PR DESCRIPTION
**Pull request by pjdufour-truss**
## Description

This PR adds some instrumentation to `CreateServiceMemberHandler`, `ShowServiceMemberHandler `, and `PatchServiceMemberHandler` so we can trace.

Also updates documentation to use `reflect.TypeOf(h).Name()` instead of manually re-writing the name of the handler.

- [x] tested on `experimental`

### CODE REVIEW

File: docs/how-to/instrument-data-in-honeycomb.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ An example of instrumenting SubmitMoveHandler, would be to to call `beeline.Star
 ```golang
 func (h SubmitMoveHandler) Handle(params moveop.SubmitMoveForApprovalParams) middleware.Responder {
     ctx := params.HTTPRequest.Context()
-    ctx, span := beeline.StartSpan(ctx, "SubmitMoveHandler")
+    ctx, span := beeline.StartSpan(ctx, reflect.TypeOf(h).Name())
```
Comment (chrisgilmerproj) [inline docs/how-to/instrument-data-in-honeycomb.md]: 👍

File: pkg/handlers/internalapi/service_members.go [hunk 2]
```diff
@@ -74,6 +83,7 @@ func (h CreateServiceMemberHandler) Handle(params servicememberop.CreateServiceM
 		var err error
 		ssn, verrs, err = models.BuildSocialSecurityNumber(ssnString.String())
 		if err != nil {
+			h.HoneyZapLogger().TraceError(ctx, "invalid social security number")
```
Comment (chrisgilmerproj) [inline pkg/handlers/internalapi/service_members.go]: This won't log the SSN will it?

Comment (pjdufour-truss) [reply pkg/handlers/internalapi/service_members.go]: No, TraceError only logs the zap fields you pass it directly.

File: pkg/handlers/internalapi/service_members.go [hunk 3]
```diff
@@ -74,6 +83,7 @@ func (h CreateServiceMemberHandler) Handle(params servicememberop.CreateServiceM
 		var err error
 		ssn, verrs, err = models.BuildSocialSecurityNumber(ssnString.String())
```
[the remark this answers is not in the document]
Comment (pjdufour-truss) [reply pkg/handlers/internalapi/service_members.go]: sure

File: pkg/handlers/internalapi/service_members.go [hunk 4]
```diff
@@ -1,16 +1,20 @@
...
 
+	"github.com/honeycombio/beeline-go"
 	"github.com/transcom/mymove/pkg/auth"
 	servicememberop "github.com/transcom/mymove/pkg/gen/internalapi/internaloperations/service_members"
 	"github.com/transcom/mymove/pkg/gen/internalmessages"
 	"github.com/transcom/mymove/pkg/handlers"
 	"github.com/transcom/mymove/pkg/models"
 	"github.com/transcom/mymove/pkg/storage"
+	"go.uber.org/zap"
+	"reflect"
```
[the remark this answers is not in the document]
Comment (pjdufour-truss) [reply pkg/handlers/internalapi/service_members.go]: Not sure what this means?  They are ordered by `go fmt`.

File: pkg/handlers/internalapi/service_members.go [hunk 5]
```diff
@@ -148,12 +166,21 @@ type ShowServiceMemberHandler struct {
...
 	serviceMemberID, _ := uuid.FromString(params.ServiceMemberID.String())
-	serviceMember, err := models.FetchServiceMemberForUser(h.DB(), session, serviceMemberID)
+
+	span.AddField("service_member_id", serviceMemberID.String())
+
+	serviceMember, err := models.FetchServiceMemberForUser(ctx, h.DB(), session, serviceMemberID)
 	if err != nil {
-		return handlers.ResponseForError(h.Logger(), err)
+		//h.HoneyZapLogger().TraceError(ctx, "Failed to fetch service member", zap.String("service_member_id", serviceMemberID.String()))
+		//return handlers.ResponseForError(h.Logger(), err)
```
Comment (dynamike) [inline pkg/handlers/internalapi/service_members.go]:
```suggestion
```

File: pkg/handlers/internalapi/service_members.go [hunk 6]
```diff
@@ -167,27 +194,44 @@ type PatchServiceMemberHandler struct {
...
 	serviceMemberID, _ := uuid.FromString(params.ServiceMemberID.String())
-	serviceMember, err := models.FetchServiceMemberForUser(h.DB(), session, serviceMemberID)
+
+	span.AddField("service_member_id", serviceMemberID.String())
+
+	serviceMember, err := models.FetchServiceMemberForUser(ctx, h.DB(), session, serviceMemberID)
 	if err != nil {
-		return handlers.ResponseForError(h.Logger(), err)
+		//h.HoneyZapLogger().TraceError(ctx, "error fetching service member", zap.String("service_member_id", params.ServiceMemberID.String()))
+		//return handlers.ResponseForError(h.Logger(), err)
```
Comment (dynamike) [inline pkg/handlers/internalapi/service_members.go]:
```suggestion
```

File: pkg/handlers/internalapi/service_members.go [hunk 7]
```diff
@@ -167,27 +194,44 @@ type PatchServiceMemberHandler struct {
...
 	}
 
 	payload := params.PatchServiceMemberPayload
-	if verrs, err := h.patchServiceMemberWithPayload(&serviceMember, payload); verrs.HasAny() || err != nil {
-		return handlers.ResponseForVErrors(h.Logger(), verrs, err)
-	}
-	if verrs, err := models.SaveServiceMember(h.DB(), &serviceMember); verrs.HasAny() || err != nil {
+	if verrs, err := h.patchServiceMemberWithPayload(ctx, &serviceMember, payload); verrs.HasAny() || err != nil {
+		// h.HoneyZapLogger().TraceError(ctx, "error patching service member", zap.String("service_member_id", serviceMember.ID.String()))
+		// return handlers.ResponseForVErrors(h.Logger(), verrs, err)
```
Comment (dynamike) [inline pkg/handlers/internalapi/service_members.go]:
```suggestion
```

File: pkg/handlers/internalapi/service_members.go [hunk 8]
```diff
@@ -275,17 +319,25 @@ type ShowServiceMemberOrdersHandler struct {
...
+
 	session := auth.SessionFromRequestContext(params.HTTPRequest)
 
 	serviceMemberID, _ := uuid.FromString(params.ServiceMemberID.String())
-	serviceMember, err := models.FetchServiceMemberForUser(h.DB(), session, serviceMemberID)
+	serviceMember, err := models.FetchServiceMemberForUser(ctx, h.DB(), session, serviceMemberID)
 	if err != nil {
-		return handlers.ResponseForError(h.Logger(), err)
+		//h.HoneyZapLogger().TraceError(ctx, "error fetching service member", zap.String("service_member_id", params.ServiceMemberID.String()))
+		//return handlers.ResponseForError(h.Logger(), err)
```
Comment (dynamike) [inline pkg/handlers/internalapi/service_members.go]:
```suggestion
```

File: pkg/handlers/internalapi/service_members.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (dynamike) [reply pkg/handlers/internalapi/service_members.go]: Usually they are groups into mymove specific packages and third party packages. If you put a new line between the two group `go fmt` should handle the rest.

**Comment by pjdufour-truss** [comment]
Tested rebased version locally.

**Pull request merged**